treeherder/etl/perf_data_adapters.py

```diff
@@ -71,61 +71,79 @@ def __init__(self):
                         "metadata": {"type": "object"}  # (holds 'options' from talos data & various auxiliary data including 'test_aux', 'talox_aux', 'results_aux', and 'results_xperf')
                     },
                     "required": [
-                        "date", "series_signature", "replicates", "testsuite",
-                        "test"
+                        "date", "series_signature", "testsuite",
                     ]
                 }
             },
             "required": ["blob", "job_guid", "name", "type"]
         }
 
-    def calculate_series_data(
-            self, job_id, result_set_id, push_timestamp, replicates):
+
+    @staticmethod
+    def _round(num):
+        # Use a precision of .2f for all numbers stored
+        # to insure we don't have floats with giant mantissa's
+        # that inflate the size of the stored data structure
+        return round(num, 2)
+
+
+    @staticmethod
+    def _calculate_summary_data(job_id, result_set_id, push_timestamp,
+                               results):
+        values = []
+        for test in results:
+            values += results[test]
+
+        if values:
+            geomean = math.exp(sum(map(math.log, values))/len(values))
```

Comment:
start reading the comments from here:
https://bugzilla.mozilla.org/show_bug.cgi?id=1021842#c24


On Thu, Feb 26, 2015 at 10:58 AM, William Lachance <notifications@github.com
> wrote:

> In treeherder/etl/perf_data_adapters.py
> <https://github.com/mozilla/treeherder-service/pull/396#discussion_r25437811>
> :
>
> > +    def _round(num):
> > +        # Use a precision of .2f for all numbers stored
> > +        # to insure we don't have floats with giant mantissa's
> > +        # that inflate the size of the stored data structure
> > +        return round(num, 2)
> > +
> > +
> > +    @staticmethod
> > +    def _calculate_summary_data(job_id, result_set_id, push_timestamp,
> > +                               results):
> > +        values = []
> > +        for test in results:
> > +            values += results[test]
> > +
> > +        if values:
> > +            geomean = math.exp(sum(map(math.log, values))/len(values))
>
> Um, maybe? :) Do you know why we did it that way in graphserver?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/treeherder-service/pull/396/files#r25437811>.
>